### DISCUSSION TITLE
Should HTML be allowed in bios?

### DISCUSSION THREAD

**Discussion by leahcimto** [theforumhelpers/theforumhelpers.github.io#247] [General]
It can be used, but should it be allowed? Or should things like bold and strikethrough be allowed but not things like changing the size of text (as it would make the page look odd)?

Please share your thoughts here.

**Comment by aidan-crockett** [comment]
If we were to filter certain tags, would this be done manually, or filtered by a parser or regex?

**Reply by leahcimto** [reply]
I think we should just reject bios that have certain tags.

**Reply by aidan-crockett** [reply]
definitely `script` or `style`, and element attributes that involve javascript, like `onclick`

**Reply by Accio1** [reply]
I think bold, italics, underline, strikethrough, and appropriate links should be allowed. All else should be declined/removed because it could mess up page styling.

**Reply by aidan-crockett** [reply]
what about `<span>`?

**Reply by Accio1** [reply]
@gosoccerboy5 I don't see when that would be useful? Doesn't it just act as inline text?

**Reply by aidan-crockett** [reply]
Like, `<span style="color: red;">`

**Reply by Accio1** [reply]
No, colouring text should not be allowed as it will make the page look weird and won't look good (or even show up properly) with all site colour schemes.

**Reply by aidan-crockett** [reply]
oh. What about `<code>`?

**Reply by Accio1** [reply]
No

**Reply by aidan-crockett** [reply]
oof

**Comment by Shluffy** [comment]
I think html in bios isn't bad. I think we should reject ones that contain anything that would mess with the style, even just putting the html in a paragraph with a custom css script as well. I think that the rule for using html should be that tags that could make the page look odd should be rejected, but a link tag is fine.

**Comment by aidan-crockett** [comment]
I'd be fine with that, I guess. We should also make sure to replace any disallowed tags with `&lt;` and `&gt;` so they appear  as, say, `<script>` instead of actually being a \<script> tag.

**Comment by Steve0Greatness** [comment]
as long as they don't contain images(that could get too big), or scripts(they could be used to download malware), sure.

**Comment by leahcimto** [comment]
Locking as Accio1 gave an answer.
